python/lsst/daf/butler/datastores/s3Datastore.py

```diff
@@ -30,14 +30,26 @@
 import pathlib
 import tempfile
 
+from botocore.exceptions import ClientError
+from http.client import ImproperConnectionState, HTTPException
+from urllib3.exceptions import RequestError, HTTPError
+
 from typing import (
     TYPE_CHECKING,
     Any,
     Optional,
     Type,
     Union,
+    Callable
 )
 
+# https://pypi.org/project/backoff/
+try:
+    import backoff
+except ImportError:
+    def backoff(func: Callable, *args: Any, **kwargs: Any) -> Callable:
```

Comment:
This doesn't quite work because the decorator is actually `backoff.on_exception`.

Reply:
oh yes, duh.

Reply:
Revised it to wrap this function in a class.